crates/oxc_linter/src/lib.rs

```diff
@@ -29,6 +37,8 @@ pub use crate::{
 pub struct Linter {
     rules: Vec<RuleEnum>,
     fix: bool,
+    timing: bool,
+    rule_timings: Arc<Mutex<FxHashMap<&'static str, RuleTimer>>>,
```

Comment:
This may not be the most appropriate data structure, but I haven't done much cross-thread stuff in Rust and I only really know about `Arc<RwLock<T>>` and `Arc<Mutex<T>>`. Let me know if there's something more suitable here.